### ISSUE TITLE
It seems like  does not support py3.8 again?

### ISSUE THREAD

**Issue by Merealtea** [huawei-noah/SMARTS#1119]
**High Level Description**
The program runs perfect until the last command  "make sanity-test"comes 
![1635996717(1)](https://user-images.githubusercontent.com/67770232/140253493-0e6073ca-34f3-4895-9315-062dc07a0ebe.png)

**Desired SMARTS version**
Version 0.4.18
**Operating System**
ubuntu 20.04 in VMware Station Player

**Problems**
I follow the guide of issue #991 and it goes wrong in last step.
The Guide is here https://github.com/huawei-noah/SMARTS/issues/991#issuecomment-874342398

**Comment by Gamenot** [comment]
@Merealtea Thank you for the question. It looks like the `SUMO_HOME` environment variable was not set. Could you confirm that `echo $SUMO_HOME` returns the path to your sumo installation?

It should look similar to:

```bash
>>> echo $SUMO_HOME
/usr/share/sumo
```

**Comment by Merealtea** [comment]
> @Merealtea Thank you for the report. It looks like the `SUMO_HOME` environment variable was not set. Could you confirm that `echo $SUMO_HOME` returns the path to your sumo installation?
> 
> It should look similar to:
> 
> ```shell
> >>> echo $SUMO_HOME
> /usr/share/sumo
> ```

Sorry,my system broke down due after I remove the python3.8. Now I changed Ubuntu version to 18.04 and installed python3.7 in it. After that I followed the guide from [official.](https://github.com/huawei-noah/SMARTS) But it still failed when I came to ”make sanity-test“. The problem  is shown below
![1636112391(1)](https://user-images.githubusercontent.com/67770232/140505198-cd95ecc8-1e26-4f8e-b000-302a7ab28f9e.png)
![1636112468(1)](https://user-images.githubusercontent.com/67770232/140505268-7371b26e-df1d-4f50-aff8-e3b2b0f7a47f.png)

**Comment by Adaickalavan** [comment]
1. Are you using the `master` branch?
2. The message: export_glb() got an unexpected keyword argument 'extras', is due to mismatched `trimesh` package version. See pr #1093 where this was addressed.
3. Could you provide a `pip list` of your current virtual environment?

**Comment by Merealtea** [comment]
> 1. Are you using the `master` branch?
> 2. The message: export_glb() got an unexpected keyword argument 'extras', is due to mismatched `trimesh` package version. See pr [Proper package versions #1093](https://github.com/huawei-noah/SMARTS/pull/1093) where this was addressed.
> 3. Could you provide a `pip list` of your current virtual environment?


 No，it seems I`m using the main branch.
![1636118310(1)](https://user-images.githubusercontent.com/67770232/140516424-99c7c84c-a68c-44ce-8b7b-23ec74c47ac2.png)
And my pip list are:
absl-py                  0.15.0
aiohttp                  3.8.0
aiohttp-cors             0.7.0
aioredis                 2.0.0
aiosignal                1.2.0
ale-py                   0.7.3
anyio                    3.3.4
argcomplete              1.12.3
argon2-cffi              21.1.0
astunparse               1.6.3
async-timeout            4.0.0
asynctest                0.13.0
atari-py                 0.2.9
attrs                    20.3.0
Automat                  20.2.0
backcall                 0.2.0
beautifulsoup4           4.10.0
bleach                   4.1.0
blessings                1.7
cached-property          1.5.2
cachetools               4.2.4
certifi                  2021.10.8
cffi                     1.15.0
charset-normalizer       2.0.7
click                    8.0.3
cloudpickle              1.3.0
colorama                 0.4.4
colorful                 0.5.4
commonmark               0.9.1
constantly               15.1.0
coverage                 6.1.1
cycler                   0.11.0
debugpy                  1.5.1
decorator                5.1.0
defusedxml               0.7.1
dm-tree                  0.1.6
entrypoints              0.3
evdev                    1.4.0
execnet                  1.9.0
filelock                 3.3.2
frozenlist               1.2.0
future                   0.18.2
gast                     0.3.3
gitdb                    4.0.9
GitPython                3.1.24
google                   3.0.0
google-api-core          2.2.2
google-auth              1.35.0
google-auth-oauthlib     0.4.6
google-pasta             0.2.0
googleapis-common-protos 1.53.0
gpustat                  0.6.0
grpcio                   1.37.0
gym                      0.21.0
h5py                     2.10.0
hyperlink                21.0.0
idna                     3.3
ijson                    3.1.4
imageio                  2.10.2
importlib-metadata       4.8.1
importlib-resources      5.4.0
incremental              21.3.0
iniconfig                1.1.1
ipykernel                6.5.0
ipython                  7.29.0
ipython-genutils         0.2.0
jedi                     0.18.0
Jinja2                   3.0.2
joblib                   1.1.0
jsonpatch                1.32
jsonpointer              2.2
jsonschema               4.2.0
jupyter-client           7.0.6
jupyter-core             4.9.1
jupyter-server           1.11.2
jupyter-server-mathjax   0.2.3
Keras-Preprocessing      1.1.2
kiwisolver               1.3.2
lz4                      3.1.3
Markdown                 3.3.4
MarkupSafe               2.0.1
matplotlib               3.4.3
matplotlib-inline        0.1.3
mistune                  0.8.4
msgpack                  1.0.2
multidict                5.2.0
nbconvert                5.6.1
nbdime                   3.1.1
nbformat                 5.1.3
nest-asyncio             1.5.1
networkx                 2.6.3
numpy                    1.18.5
nvidia-ml-py3            7.352.0
oauthlib                 3.1.1
opencensus               0.8.0
opencensus-context       0.1.2
opencv-python-headless   4.3.0.36
opt-einsum               3.3.0
packaging                21.2
pandas                   1.3.4
pandocfilters            1.5.0
parso                    0.8.2
pexpect                  4.8.0
pickleshare              0.7.5
Pillow                   8.4.0
pip                      21.3.1
pkg_resources            0.0.0
pluggy                   1.0.0
prometheus-client        0.12.0
prompt-toolkit           3.0.22
protobuf                 3.19.1
psutil                   5.8.0
ptyprocess               0.7.0
py                       1.11.0
py-cpuinfo               8.0.0
py-spy                   0.3.10
pyasn1                   0.4.8
pyasn1-modules           0.2.8
pybullet                 3.0.6
pycparser                2.20
Pygments                 2.10.0
pynput                   1.7.4
pyparsing                2.4.7
pyrsistent               0.18.0
pytest                   6.2.5
pytest-benchmark         3.4.1
pytest-cov               3.0.0
pytest-forked            1.3.0
pytest-notebook          0.6.1
pytest-xdist             2.4.0
python-dateutil          2.8.2
python-xlib              0.31
pytz                     2021.3
PyWavelets               1.1.1
PyYAML                   6.0
pyzmq                    22.3.0
ray                      1.0.1.post1
redis                    3.4.1
requests                 2.26.0
requests-oauthlib        1.3.0
rich                     10.12.0
rsa                      4.7.2
Rtree                    0.9.7
scikit-image             0.18.3
scikit-learn             1.0.1
scipy                    1.4.1
Send2Trash               1.8.0
setuptools               58.5.3
sh                       1.14.2
Shapely                  1.8.0
six                      1.16.0
sklearn                  0.0
smarts                   0.4.18      /home/cxy/NGSIM_SMARTS/SMARTS
smmap                    5.0.0
sniffio                  1.2.0
soupsieve                2.3
tableprint               0.9.1
tabulate                 0.8.9
tensorboard              2.2.2
tensorboard-plugin-wit   1.8.0
tensorboardX             2.4
tensorflow               2.2.1
tensorflow-estimator     2.2.0
termcolor                1.1.0
terminado                0.12.1
testpath                 0.5.0
threadpoolctl            3.0.0
tifffile                 2021.11.2
toml                     0.10.2
tomli                    1.2.2
torch                    1.4.0
torchfile                0.1.0
torchvision              0.5.0
tornado                  6.1
traitlets                5.1.1
trimesh                  3.9.35
Twisted                  21.7.0
typing-extensions        3.10.0.2
urllib3                  1.26.7
visdom                   0.1.8.9
wcwidth                  0.2.5
webencodings             0.5.1
websocket-client         1.2.1
Werkzeug                 2.0.2
wheel                    0.37.0
wrapt                    1.13.3
yarl                     1.7.2
yattag                   1.14.0
zipp                     3.6.0
zope.interface           5.4.0
I install all these using `pip install -r requirements.txt`

**Comment by Merealtea** [comment]
> 1. Are you using the `master` branch?
> 2. The message: export_glb() got an unexpected keyword argument 'extras', is due to mismatched `trimesh` package version. See pr [Proper package versions #1093](https://github.com/huawei-noah/SMARTS/pull/1093) where this was addressed.
> 3. Could you provide a `pip list` of your current virtual environment?

NO, It seems I`m using the main branch.
![1636118335](https://user-images.githubusercontent.com/67770232/140517169-7486dbe6-2516-4f35-b7b4-f934cc5e2eb6.png)

And these are my 'pip list':

absl-py                  0.15.0
aiohttp                  3.8.0
aiohttp-cors             0.7.0
aioredis                 2.0.0
aiosignal                1.2.0
ale-py                   0.7.3
anyio                    3.3.4
argcomplete              1.12.3
argon2-cffi              21.1.0
astunparse               1.6.3
async-timeout            4.0.0
asynctest                0.13.0
atari-py                 0.2.9
attrs                    20.3.0
Automat                  20.2.0
backcall                 0.2.0
beautifulsoup4           4.10.0
bleach                   4.1.0
blessings                1.7
cached-property          1.5.2
cachetools               4.2.4
certifi                  2021.10.8
cffi                     1.15.0
charset-normalizer       2.0.7
click                    8.0.3
cloudpickle              1.3.0
colorama                 0.4.4
colorful                 0.5.4
commonmark               0.9.1
constantly               15.1.0
coverage                 6.1.1
cycler                   0.11.0
debugpy                  1.5.1
decorator                5.1.0
defusedxml               0.7.1
dm-tree                  0.1.6
entrypoints              0.3
evdev                    1.4.0
execnet                  1.9.0
filelock                 3.3.2
frozenlist               1.2.0
future                   0.18.2
gast                     0.3.3
gitdb                    4.0.9
GitPython                3.1.24
google                   3.0.0
google-api-core          2.2.2
google-auth              1.35.0
google-auth-oauthlib     0.4.6
google-pasta             0.2.0
googleapis-common-protos 1.53.0
gpustat                  0.6.0
grpcio                   1.37.0
gym                      0.21.0
h5py                     2.10.0
hyperlink                21.0.0
idna                     3.3
ijson                    3.1.4
imageio                  2.10.2
importlib-metadata       4.8.1
importlib-resources      5.4.0
incremental              21.3.0
iniconfig                1.1.1
ipykernel                6.5.0
ipython                  7.29.0
ipython-genutils         0.2.0
jedi                     0.18.0
Jinja2                   3.0.2
joblib                   1.1.0
jsonpatch                1.32
jsonpointer              2.2
jsonschema               4.2.0
jupyter-client           7.0.6
jupyter-core             4.9.1
jupyter-server           1.11.2
jupyter-server-mathjax   0.2.3
Keras-Preprocessing      1.1.2
kiwisolver               1.3.2
lz4                      3.1.3
Markdown                 3.3.4
MarkupSafe               2.0.1
matplotlib               3.4.3
matplotlib-inline        0.1.3
mistune                  0.8.4
msgpack                  1.0.2
multidict                5.2.0
nbconvert                5.6.1
nbdime                   3.1.1
nbformat                 5.1.3
nest-asyncio             1.5.1
networkx                 2.6.3
numpy                    1.18.5
nvidia-ml-py3            7.352.0
oauthlib                 3.1.1
opencensus               0.8.0
opencensus-context       0.1.2
opencv-python-headless   4.3.0.36
opt-einsum               3.3.0
packaging                21.2
pandas                   1.3.4
pandocfilters            1.5.0
parso                    0.8.2
pexpect                  4.8.0
pickleshare              0.7.5
Pillow                   8.4.0
pip                      21.3.1
pkg_resources            0.0.0
pluggy                   1.0.0
prometheus-client        0.12.0
prompt-toolkit           3.0.22
protobuf                 3.19.1
psutil                   5.8.0
ptyprocess               0.7.0
py                       1.11.0
py-cpuinfo               8.0.0
py-spy                   0.3.10
pyasn1                   0.4.8
pyasn1-modules           0.2.8
pybullet                 3.0.6
pycparser                2.20
Pygments                 2.10.0
pynput                   1.7.4
pyparsing                2.4.7
pyrsistent               0.18.0
pytest                   6.2.5
pytest-benchmark         3.4.1
pytest-cov               3.0.0
pytest-forked            1.3.0
pytest-notebook          0.6.1
pytest-xdist             2.4.0
python-dateutil          2.8.2
python-xlib              0.31
pytz                     2021.3
PyWavelets               1.1.1
PyYAML                   6.0
pyzmq                    22.3.0
ray                      1.0.1.post1
redis                    3.4.1
requests                 2.26.0
requests-oauthlib        1.3.0
rich                     10.12.0
rsa                      4.7.2
Rtree                    0.9.7
scikit-image             0.18.3
scikit-learn             1.0.1
scipy                    1.4.1
Send2Trash               1.8.0
setuptools               58.5.3
sh                       1.14.2
Shapely                  1.8.0
six                      1.16.0
sklearn                  0.0
smarts                   0.4.18      /home/cxy/NGSIM_SMARTS/SMARTS
smmap                    5.0.0
sniffio                  1.2.0
soupsieve                2.3
tableprint               0.9.1
tabulate                 0.8.9
tensorboard              2.2.2
tensorboard-plugin-wit   1.8.0
tensorboardX             2.4
tensorflow               2.2.1
tensorflow-estimator     2.2.0
termcolor                1.1.0
terminado                0.12.1
testpath                 0.5.0
threadpoolctl            3.0.0
tifffile                 2021.11.2
toml                     0.10.2
tomli                    1.2.2
torch                    1.4.0
torchfile                0.1.0
torchvision              0.5.0
tornado                  6.1
traitlets                5.1.1
trimesh                  3.9.35
Twisted                  21.7.0
typing-extensions        3.10.0.2
urllib3                  1.26.7
visdom                   0.1.8.9
wcwidth                  0.2.5
webencodings             0.5.1
websocket-client         1.2.1
Werkzeug                 2.0.2
wheel                    0.37.0
wrapt                    1.13.3
yarl                     1.7.2
yattag                   1.14.0
zipp                     3.6.0
zope.interface           5.4.0

I installed all these using 'pip install -r requirements.txt'

**Comment by Adaickalavan** [comment]
1. From the pip list, it appears that we have installed `trimesh==3.9.35` and `aiohttp==3.8.0`, which is different from the package version `trimesh==3.9.20` and `aiohttp==3.7.4.post0` in `requirements.txt` file.
2. To ensure pip list matches the requirements.txt, try pip install with `--no-cache-dir` option.
    ```
    $ cd SMARTS/
    $ python3.7 -m venv .venv
    $ source .venv/bin/activate
    $ pip install --upgrade pip
    $ pip install --no-cache-dir -r requirements.txt
    ```

**Comment by Merealtea** [comment]
> 1. From the pip list, it appears that we have installed `trimesh==3.9.35` and `aiohttp==3.8.0`, which is different from the package version `trimesh==3.9.20` and `aiohttp==3.7.4.post0` in `requirements.txt` file.
> 2. To ensure pip list matches the requirements.txt, try pip install with `--no-cache-dir` option.
>    ```
>    $ cd SMARTS/
>    $ python3.7 -m venv .venv
>    $ source .venv/bin/activate
>    $ pip install --upgrade pip
>    $ pip install --no-cache-dir -r requirements.txt
>    ```

Yes! It has been solved wonderfully. My simulation can run properly! Thank U!
